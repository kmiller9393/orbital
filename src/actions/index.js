export const addGlossary = glossary => ({
  type: 'ADD_GLOSSARY',
  glossary
});

export const addTriviaSet = triviaSet => ({
  type: 'ADD_TRIVIA_SET',
  triviaSet
});

export const addScore = score => ({
  type: 'ADD_SCORE',
  score
});
