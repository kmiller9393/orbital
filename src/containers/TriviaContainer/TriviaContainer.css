.trivia-container-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px; }

.trivia-title {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif; }

.trivia-container-header {
  padding-right: 6.5rem; }

.back-arrow {
  height: 40px;
  padding: 0rem 4rem 0.5rem 0rem; }

.answers-container {
  color: #f7f7f7;
  padding: 0 0.75rem 0 0.75rem; }

.trivia-card {
  align-items: center;
  background-color: #b72424;
  border-radius: 8px;
  cursor: pointer;
  display: flex;
  font-size: 1.1rem;
  height: 70px;
  justify-content: center;
  margin: 1.5rem 1rem 1rem 1rem; }
