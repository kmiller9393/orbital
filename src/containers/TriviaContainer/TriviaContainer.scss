.trivia-container-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px;
}

.trivia-title {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
}

.answers-container {
  color: #f7f7f7;
  padding: 0 0.75rem 0 0.75rem;
}

.trivia-card {
  align-items: center;
  background-color: #b72424;
  border-radius: 8px;
  display: flex;
  font-size: 1.1rem;
  height: 70px;
  justify-content: center;
  margin: 1.5rem 1rem 1rem 1rem;
}
