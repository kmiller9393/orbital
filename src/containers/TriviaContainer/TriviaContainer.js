import React, { Component } from 'react';
import TriviaCard from '../TriviaCard/TriviaCard';
import { NavLink, withRouter } from 'react-router-dom';
import {
  addTriviaOne,
  addTriviaTwo,
  addTriviaThree,
  addTriviaFour,
  addTriviaFive
} from '../../utils/helper';
import { connect } from 'react-redux';
import { addScore } from '../../actions';
import PropTypes from 'prop-types';
import orbital from '../../images/orbital-header.svg';
import './TriviaContainer.css';

class TriviaContainer extends Component {
  constructor() {
    super();
    this.state = {
      page: 0,
      triviaItems: [],
      ready: false
    };
  }

  componentDidMount() {
    const { triviaAnswers, history } = this.props;
    const { ready } = this.state;
    let triviaItems;

    if (history.location.pathname.includes('/trivia-one')) {
      triviaItems = addTriviaOne(triviaAnswers);
      this.setState({ triviaItems, ready: !ready });
    }

    if (history.location.pathname.includes('/trivia-two')) {
      triviaItems = addTriviaTwo(triviaAnswers);
      this.setState({ triviaItems, ready: !ready });
    }

    if (history.location.pathname.includes('/trivia-three')) {
      triviaItems = addTriviaThree(triviaAnswers);
      this.setState({ triviaItems, ready: !ready });
    }

    if (history.location.pathname.includes('/trivia-four')) {
      triviaItems = addTriviaFour(triviaAnswers);
      this.setState({ triviaItems, ready: !ready });
    }

    if (history.location.pathname.includes('/trivia-five')) {
      triviaItems = addTriviaFive(triviaAnswers);
      this.setState({ triviaItems, ready: !ready });
    }
  }

  checkAnswer = e => {
    const { history, addScore, score } = this.props;
    const { page, triviaItems } = this.state;

    const answers = triviaItems.map(subject => subject.answer);
    let newPage = page;

    if (e.target.innerText === answers[newPage]) {
      addScore(score);
    }

    this.setState({ page: newPage + 1 });

    if (history.location.pathname.includes('/trivia-one')) {
      history.replace(`/trivia-one/${[newPage]}`);
    }

    if (history.location.pathname.includes('/trivia-two')) {
      history.replace(`/trivia-two/${[newPage]}`);
    }
    if (history.location.pathname.includes('/trivia-three')) {
      history.replace(`/trivia-three/${[newPage]}`);
    }

    if (history.location.pathname.includes('/trivia-four')) {
      history.replace(`/trivia-four/${[newPage]}`);
    }

    if (history.location.pathname.includes('/trivia-five')) {
      history.replace(`/trivia-five/${[newPage]}`);
    }
  };

  render() {
    const { page, triviaItems, ready } = this.state;
    const { triviaAnswers } = this.props;
    const answers = triviaAnswers.map(subject => {
      return <TriviaCard {...subject} key={subject.id} />;
    });

    return (
      <div>
        <NavLink to="/">
          <img className="trivia-container-image" src={orbital} alt="Orbital" />
        </NavLink>
        <h2 className="trivia-title">Weekly Trivia</h2>

        <div className="answers-container">
          {ready && <h3>What is {triviaItems[page].question}?</h3>}
          <div onClick={e => this.checkAnswer(e)}>
            {answers[Math.floor(Math.random() * 46)]}
          </div>
          {ready && (
            <div onClick={e => this.checkAnswer(e)} className="trivia-card">
              {triviaItems[page].answer}
            </div>
          )}
          <div onClick={e => this.checkAnswer(e)}>
            {answers[Math.floor(Math.random() * 46)]}
          </div>
          <div onClick={e => this.checkAnswer(e)}>
            {answers[Math.floor(Math.random() * 46)]}
          </div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  triviaAnswers: state.triviaAnswers,
  score: state.score
});

export const mapDispatchToProps = dispatch => ({
  addScore: score => dispatch(addScore(score))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(TriviaContainer)
);

const { array, object, func, number } = PropTypes;
TriviaContainer.propTypes = {
  triviaAnswers: array,
  history: object,
  addScore: func,
  score: number
};
