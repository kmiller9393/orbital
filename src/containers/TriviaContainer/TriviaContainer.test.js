import TriviaContainer, {
  mapStateToProps,
  mapDispatchToProps
} from './TriviaContainer';
import { addScore } from '../../actions';

describe('.TriviaContainer', () => {
  describe('mapStateToProps', () => {
    it('should return an array of triviaAnswer objects and the score', () => {
      const question = 'a small solar system object composed mostly of rock';
      const answer = 'Asteroid';
      const id = 0;

      const mockState = {
        triviaAnswers: [{ question, answer, id }],
        score: 0
      };

      const expected = {
        triviaAnswers: [{ question, answer, id }],
        score: 0
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addScore(0);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addScore(0);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
