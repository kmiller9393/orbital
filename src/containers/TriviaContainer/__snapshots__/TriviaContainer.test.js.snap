// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaContainer TriviaContainer component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TriviaContainer
    history={
        Object {
            "location": Object {
              "pathname": "/review-one",
            },
            "replace": [Function],
          }
    }
    location={
        Object {
            "pathname": "/review-one",
          }
    }
    triviaAnswers={
        Array [
            Object {
              "answer": "Asteroid",
              "id": 0,
              "question": "a small solar system object composed mostly of rock\\"",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="trivia-container-header"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="trivia-center"
          >
                    <img
                              alt="Navigate Back"
                              className="back-arrow"
                              src="left-arrow.svg"
                    />
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
          >
                    <img
                              alt="Orbital"
                              className="trivia--image-after"
                              src="orbital-header.svg"
                    />
          </NavLink>
</div>,
        <h2
          className="trivia-title"
>
          Weekly Trivia
</h2>,
        <div
          className="answers-container"
>
          <div
                    className="choice-card"
                    onClick={[Function]}
          />
          <div
                    onClick={[Function]}
          />
          <div
                    onClick={[Function]}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="trivia-center"
>
              <img
                            alt="Navigate Back"
                            className="back-arrow"
                            src="left-arrow.svg"
              />
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
>
              <img
                            alt="Orbital"
                            className="trivia--image-after"
                            src="orbital-header.svg"
              />
</NavLink>,
            false,
          ],
          "className": "trivia-container-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <img
                alt="Navigate Back"
                className="back-arrow"
                src="left-arrow.svg"
/>,
              "to": "trivia-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Navigate Back",
                "className": "back-arrow",
                "src": "left-arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <img
                alt="Orbital"
                className="trivia--image-after"
                src="orbital-header.svg"
/>,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Orbital",
                "className": "trivia--image-after",
                "src": "orbital-header.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Weekly Trivia",
          "className": "trivia-title",
        },
        "ref": null,
        "rendered": "Weekly Trivia",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div
              className="choice-card"
              onClick={[Function]}
/>,
            false,
            <div
              onClick={[Function]}
/>,
            <div
              onClick={[Function]}
/>,
          ],
          "className": "answers-container",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "choice-card",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="trivia-container-header"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="trivia-center"
            >
                        <img
                                    alt="Navigate Back"
                                    className="back-arrow"
                                    src="left-arrow.svg"
                        />
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/"
            >
                        <img
                                    alt="Orbital"
                                    className="trivia--image-after"
                                    src="orbital-header.svg"
                        />
            </NavLink>
</div>,
          <h2
            className="trivia-title"
>
            Weekly Trivia
</h2>,
          <div
            className="answers-container"
>
            <div
                        className="choice-card"
                        onClick={[Function]}
            />
            <div
                        onClick={[Function]}
            />
            <div
                        onClick={[Function]}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="trivia-center"
>
                <img
                                alt="Navigate Back"
                                className="back-arrow"
                                src="left-arrow.svg"
                />
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
>
                <img
                                alt="Orbital"
                                className="trivia--image-after"
                                src="orbital-header.svg"
                />
</NavLink>,
              false,
            ],
            "className": "trivia-container-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <img
                  alt="Navigate Back"
                  className="back-arrow"
                  src="left-arrow.svg"
/>,
                "to": "trivia-center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Navigate Back",
                  "className": "back-arrow",
                  "src": "left-arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <img
                  alt="Orbital"
                  className="trivia--image-after"
                  src="orbital-header.svg"
/>,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Orbital",
                  "className": "trivia--image-after",
                  "src": "orbital-header.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weekly Trivia",
            "className": "trivia-title",
          },
          "ref": null,
          "rendered": "Weekly Trivia",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <div
                className="choice-card"
                onClick={[Function]}
/>,
              false,
              <div
                onClick={[Function]}
/>,
              <div
                onClick={[Function]}
/>,
            ],
            "className": "answers-container",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "choice-card",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
